modules/react/text-area/lib/TextArea.tsx

```diff
@@ -37,71 +38,59 @@ export interface TextAreaProps extends Themeable, GrowthBehavior {
    * The resize constraints of the TextArea.
    * @default TextArea.ResizeDirection.Both
    */
-  resize?: TextAreaResizeDirection;
+  resize?: ValueOf<typeof TextAreaResizeDirection>;
   /**
    * The value of the TextArea.
    */
   value?: any;
 }
 
-export enum TextAreaResizeDirection {
-  None = 'none',
-  Both = 'both',
-  Horizontal = 'horizontal',
-  Vertical = 'vertical',
-}
+export const TextAreaResizeDirection = {
+  None: 'none' as 'none',
+  Both: 'both' as 'both',
+  Horizontal: 'horizontal' as 'horizontal',
+  Vertical: 'vertical' as 'vertical',
+};
 
-const StyledTextArea = styled('textarea')<TextAreaProps & StyledType>(
-  ({theme, error}) => ({
-    ...type.levels.subtext.large,
-    border: `1px solid ${inputColors.border}`,
-    display: 'block',
-    backgroundColor: inputColors.background,
-    borderRadius: borderRadius.m,
-    boxSizing: 'border-box',
-    minHeight: space.xxl,
-    minWidth: `calc((${space.xxxl} * 3) + ${space.xl})`,
-    transition: '0.2s box-shadow, 0.2s border-color',
-    padding: space.xxs, // Compensate for border
-    margin: 0, // Fix Safari
+export const textAreaStencil = createStencil({
+  extends: textInputStencil,
+  base: {
+    minHeight: system.space.x16,
+    minWidth: calc.add(calc.multiply(system.space.x20, 3), system.space.x10),
     '&::webkit-resizer': {
       display: 'none',
     },
-    '&::placeholder': {
-      color: inputColors.placeholder,
-    },
-    '&:hover': {
-      borderColor: inputColors.hoverBorder,
-    },
-    '&:focus:not([disabled])': {
-      borderColor: theme.canvas.palette.common.focusOutline,
-      boxShadow: `inset 0 0 0 1px ${theme.canvas.palette.common.focusOutline}`,
-      outline: 'none',
-    },
-    '&:disabled': {
-      backgroundColor: inputColors.disabled.background,
-      borderColor: inputColors.disabled.border,
-      color: inputColors.disabled.text,
-      '&::placeholder': {
-        color: inputColors.disabled.text,
+  },
+  modifiers: {
+    resize: {
+      both: {
+        resize: 'both',
+      },
+      horizontal: {
+        resize: 'horizontal',
+      },
+      vertical: {
+        resize: 'vertical',
+      },
+      none: {
+        resize: 'none',
```

Comment:
Can't it be variable as modifier name and value is the same?

Reply:
I'll try that!

Reply:
hmm, nor sure what you mean, do you have an example or something I can reference?

Reply:
i mean something like this:

```tsx
export const textAreaStencil = createStencil({
  extends: textInputStencil,
  vars: {
    resize: 'both'
  },
  base: ({resize}) => ({
    minHeight: system.space.x16,
    minWidth: calc.add(calc.multiply(system.space.x20, 3), system.space.x10),
    resize,
    '&::webkit-resizer': {
      display: 'none',
    },
  }),
}

```

Reply:
I'm not sure if Raisa is asking if `resize` can be a variable instead of a modifier. It doesn't make much difference to the Stencil API or the component:

```ts
// with variable
textAreaStencil({ resize: 'both' });
<TextArea resize="both" />

// with modifier
textAreaStencil({ resize: 'both' });
<TextArea resize="both" />
```

The difference is in CSS:
```css
/* with variable */
<textarea class="cnvs-text-area" style="--cnvs-text-area-resize: both;" />

/* with modifier */
<textarea class="cnvs-text-area resize-both" />
```

A difference is in the allowed interface. The modifier has a limited amount of input it can accept while the variable has no restriction. As far as which is better for CSS? The modifier seems more concise.

Reply:
Hmm fair, when I change this to a variable i get a typescript issue unrelated to the stencil